### PR TITLE
Add automatic build image push to quay.io

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
This adds a master GitHub action which updates the build image on every
push.
#### Which issue(s) this PR fixes:

None
#### Does this PR have test?

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

/hold for some manual test

### CODE REVIEW

File: hack/build-image/Dockerfile [hunk 1]
```diff
@@ -36,9 +36,9 @@ ENV ENV=/etc/profile \
     NIX_SSL_CERT_FILE=/etc/ssl/certs/ca-certificates.crt \
     NIX_PATH=/nix/var/nix/profiles/per-user/root/channels
 
+RUN nix-env -iA cachix -f https://cachix.org/api/v1/install && \
+    cachix use security-profiles-operator
+
 RUN nix-collect-garbage --delete-old -d && \
     nix-store --optimise && \
     nix-store --verify --check-contents
```
Comment (saschagrunert) [inline hack/build-image/Dockerfile:44]: Let's cleanup the store after cachix has been installed, makes more sense.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -25,3 +25,18 @@ jobs:
           name: security-profiles-operator
           authToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
       - run: make nix nix-arm64
+
+  image:
+    runs-on: ubuntu-latest
+    env:
+      USER: security-profiles-operator+github
+      IMAGE: quay.io/security-profiles-operator/build
+      TOKEN: ${{ secrets.QUAY_TOKEN }}
```
Comment (saschagrunert) [inline .github/workflows/build.yml:34]: The token is already set within the repository, so the master run should push a new image `latest`.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -25,3 +25,18 @@ jobs:
...
+    steps:
+      - uses: actions/checkout@v2
+      - run: sudo podman build hack/build-image -t $IMAGE
+      - name: Push the image
+        if: github.ref == 'refs/heads/master'
+        run: |
+          sudo podman login -u $USER -p $TOKEN
+          sudo podman push $IMAGE
```
Comment (JAORMX) [inline .github/workflows/build.yml:42]: This looks fine; though we could use the buildah action instead https://github.com/marketplace/actions/buildah-build

then the push to registry action https://github.com/redhat-actions/push-to-registry

Comment (saschagrunert) [reply .github/workflows/build.yml]: Sounds good, using them now.

**Comment by saschagrunert** [comment]
@JAORMX PTAL again

**Pull request merged**